lib/logstash/inputs/snmp.rb

```diff
@@ -128,112 +129,115 @@ def register
     end
 
     @client = build_client!(mib_manager, supported_transports, hosts_versions)
+    @request_aggregator = create_request_aggregator
   end
 
   def run(queue)
+    # Put the client into the listen mode, so it can send requests and receive responses
     @client.listen
-    # for now a naive single threaded poller which sleeps off the remaining interval between
-    # each run. each run polls all the defined hosts for the get, table and walk options.
-    stoppable_interval_runner.every(@interval, "polling hosts") do
+
+    # Single threaded poller which sleeps off the remaining interval between each run.
+    # Each run polls all the defined hosts for the get, table and walk options.
+    stoppable_interval_runner.every(@interval, 'polling hosts') do
```

Comment:
In the future we'll want to track the interval per host instead of globally. otherwise a list of 100 hosts will be severely impacted by a few timing out or being slow, even with multiple threads.
That said, things are much better now compared to the previous implementation :)